### PR TITLE
Network isolation validation

### PR DESCRIPTION
**Pull request by robertvolkmann**
Validate network isolation configuration in a single location rather than performing validation in multiple places

### CODE REVIEW

**Comment by robertvolkmann** [comment]
@vknabel Do want to support empty RegistryMirrors or not?
```
	if p.NetworkIsolation == nil || len(p.NetworkIsolation.RegistryMirrors) == 0 {
		m.logger.Info("no registry mirrors specified in this shoot, nothing to do here", "shoot", shootName)
		return nil
	}
```
Source: https://github.com/metal-stack/gardener-extension-provider-metal/blob/b38f4e809a4fd20ae5d0b48022f20784a1e32521/pkg/webhook/shoot/mutator.go#L170

**Comment by vknabel** [comment]
@robertvolkmann if the network access type is not baseline, the `NetworkIsolation` and its `RegistryMirrors` must be set and contain at least one mirror.

File: pkg/apis/metal/validation/network_access.go [hunk 1]
```diff
@@ -32,46 +32,9 @@ func validateNetworkAccessFields(controlPlaneConfig *apismetal.ControlPlaneConfi
...
-			partNiPath.Child("allowedNetworks", "egress"),
-			partition.NetworkIsolation.AllowedNetworks.Egress,
-			"may not be empty",
-		))
-	}
-	if len(partition.NetworkIsolation.AllowedNetworks.Ingress) == 0 {
-		allErrs = append(allErrs, field.Invalid(
-			partNiPath.Child("allowedNetworks", "ingress"),
-			partition.NetworkIsolation.AllowedNetworks.Ingress,
-			"may not be empty",
```
Comment (vknabel) [inline pkg/apis/metal/validation/network_access.go:73]: This part is actually important. Creating an isolated shoot should actually require DNS, NTP, mirrors and allowed networks. Nodes will not be able to come up and will leave the shoot in a broken state.

**Review by vknabel — APPROVED** [review]

**Comment by majst01** [comment]
rebase and run in test

**Comment by vknabel** [comment]
Worked fine in test

**Pull request merged**